### DISCUSSION TITLE
Building apps with multiple webviews

### DISCUSSION THREAD

**Discussion by kishaningithub** [neutralinojs/neutralinojs#806] [Q&A]
What does it take to build something like https://getferdi.com/ in Neutralino?

Are webviews(whatsapp, gmail etc) within webview(parent application) supported?

Alternate ways to build that in Neutralino without using multiple webviews are also welcome

PS - I am very new to Neutralino.

**Comment by pegvin** [comment]
I Won't suggest this, since Neutralino Uses Webview already in the os, but there maybe times where the user is using windows 7 or even older version of windows which doesn't have the latest edge version which may cause the website to not work.

because many APIs aren't implemented in old edge versions

**Reply by kishaningithub** [reply]
I agree, say if i restrict the app only to run in the latest versions of Operating systems so that latest webview is always used.

Is there a way in Neutralino to orchestrate multiple web views each having their own web sessions?

**Reply by pegvin** [reply]
Are you Talking About https://neutralino.js.org/docs/api/window/#windowcreateurl-windowoptions ?

**Reply by kishaningithub** [reply]
Not exactly though.. what is wanted is an ability to swap out webviews within the same window(screenshot below)


If you see the below screenshot, when i click on a service(gmail, Calendar, whatsapp etc) the webview switches to load that particular page, also the sessions are maintained even after an application restart and the sessions are fully isolated from each other.

This is what i am wondering if this can be possible using neutralinojs as of now or in the future...

### Screenshot
<img width="600" alt="Screenshot 2022-03-03 at 10 48 30 AM" src="https://user-images.githubusercontent.com/763760/156501221-df7123d5-ac6d-4e35-b773-7f33b9fd3225.png">

**Reply by pegvin** [reply]
currently no, there is no way of doing that, maybe in future we get something?

**Reply by niutech** [reply]
+1 for implementing something like [BrowserView](https://www.electronjs.org/docs/latest/api/browser-view) from Electron.

**Comment by siniradam** [comment]
Would be immensely helpful to build apps similar to Franz, or to create a hub for multiple control panels.

**Reply by lucasengithub** [reply]
i was searching for it
